kie-server-parent/kie-server-services/kie-server-services-scenario-simulation/src/main/java/org/kie/server/services/scenariosimulation/ScenarioSimulationKieServerExtension.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.server.services.scenariosimulation;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+import org.kie.server.api.KieServerConstants;
+import org.kie.server.api.model.Message;
+import org.kie.server.api.model.Severity;
+import org.kie.server.services.api.KieContainerInstance;
+import org.kie.server.services.api.KieServerApplicationComponentsService;
+import org.kie.server.services.api.KieServerExtension;
+import org.kie.server.services.api.KieServerRegistry;
+import org.kie.server.services.api.SupportedTransports;
+import org.kie.server.services.impl.KieServerImpl;
+
+public class ScenarioSimulationKieServerExtension implements KieServerExtension {
+
+    public static final String EXTENSION_NAME = "Scenario Simulation";
+
+    private static final Boolean SCESIM_DISABLED = Boolean.parseBoolean(System.getProperty(KieServerConstants.KIE_SCENARIO_SIMULATION_SERVER_EXT_DISABLED, "true"));
+    private static final Boolean DROOLS_DISABLED = Boolean.parseBoolean(System.getProperty(KieServerConstants.KIE_DROOLS_SERVER_EXT_DISABLED, "false"));
+    private static final Boolean DMN_DISABLED = Boolean.parseBoolean(System.getProperty(KieServerConstants.KIE_DMN_SERVER_EXT_DISABLED, "false"));
+
+    private KieServerRegistry context;
+    private ScenarioSimulationService scenarioSimulationService;
+    private boolean initialized = false;
+
+    @Override
+    public boolean isInitialized() {
+        return initialized;
+    }
+
+    /**
+     * This extension is active if it is not disabled and one of DMN or Drools is active
+     * @return
+     */
+    @Override
+    public boolean isActive() {
+        return !SCESIM_DISABLED && (!DROOLS_DISABLED || !DMN_DISABLED);
```

Comment:
@danielezonca 
a negation over an sc `AND` and the result of an sc `OR` over two negations: is there a way to simplify it ?

Reply:
Improved comment and method. I cannot really simplify it because all kie-server extensions are created with this `not disabled = enabled` logic that is not really readable.